File: samples/client/3_0_3_unit_test/java/src/test/java/org/openapijsonschematools/components/schemas/IntegerTypeMatchesIntegersTest.java
package org.openapijsonschematools.components.schemas;

import org.junit.Assert;
import org.junit.Test;
import org.openapijsonschematools.configurations.JsonSchemaKeywordFlags;
import org.openapijsonschematools.configurations.SchemaConfiguration;
import org.openapijsonschematools.exceptions.ValidationException;

import java.util.List;
import java.util.Map;
import java.util.AbstractMap;

public class IntegerTypeMatchesIntegersTest {
    static final SchemaConfiguration configuration = new SchemaConfiguration(JsonSchemaKeywordFlags.ofNone());

    @Test
    public void testAnObjectIsNotAnIntegerFails() {
        // an object is not an integer
        Assert.assertThrows(ValidationException.class, () -> IntegerTypeMatchesIntegers.IntegerTypeMatchesIntegers1.validate(
            Map.ofEntries(
            ),
            configuration
        ));
    }

    @Test
    public void testAnArrayIsNotAnIntegerFails() {
        // an array is not an integer
        Assert.assertThrows(ValidationException.class, () -> IntegerTypeMatchesIntegers.IntegerTypeMatchesIntegers1.validate(
            List.of(
            ),
            configuration
        ));
    }

    @Test
    public void testNullIsNotAnIntegerFails() {
        // null is not an integer
        Assert.assertThrows(ValidationException.class, () -> IntegerTypeMatchesIntegers.IntegerTypeMatchesIntegers1.validate(
            null,
            configuration
        ));
    }

    @Test
    public void testAFloatWithZeroFractionalPartIsAnIntegerPasses() {
        // a float with zero fractional part is an integer
        IntegerTypeMatchesIntegers.IntegerTypeMatchesIntegers1.validate(
            1.0,
            configuration
        );
    }

    @Test
    public void testABooleanIsNotAnIntegerFails() {
        // a boolean is not an integer
        Assert.assertThrows(ValidationException.class, () -> IntegerTypeMatchesIntegers.IntegerTypeMatchesIntegers1.validate(
            true,
            configuration
        ));
    }

    @Test
    public void testAStringIsStillNotAnIntegerEvenIfItLooksLikeOneFails() {
        // a string is still not an integer, even if it looks like one
        Assert.assertThrows(ValidationException.class, () -> IntegerTypeMatchesIntegers.IntegerTypeMatchesIntegers1.validate(
            "1",
            configuration
        ));
    }

    @Test
    public void testAStringIsNotAnIntegerFails() {
        // a string is not an integer
        Assert.assertThrows(ValidationException.class, () -> IntegerTypeMatchesIntegers.IntegerTypeMatchesIntegers1.validate(
            "foo",
            configuration
        ));
    }

    @Test
    public void testAnIntegerIsAnIntegerPasses() {
        // an integer is an integer
        IntegerTypeMatchesIntegers.IntegerTypeMatchesIntegers1.validate(
            1,
            configuration
        );
    }

    @Test
    public void testAFloatIsNotAnIntegerFails() {
        // a float is not an integer
        Assert.assertThrows(ValidationException.class, () -> IntegerTypeMatchesIntegers.IntegerTypeMatchesIntegers1.validate(
            1.1,
            configuration
        ));
    }
}

File: samples/client/3_1_0_unit_test/kotlin/src/test/kotlin/org/openapijsonschematools/client/components/schemas/IntegerTypeMatchesIntegersTest.kt
package org.openapijsonschematools.client.components.schemas

import org.openapijsonschematools.client.configurations.JsonSchemaKeywordFlags
import org.openapijsonschematools.client.configurations.SchemaConfiguration
import org.openapijsonschematools.client.exceptions.ValidationException

import kotlin.test.Test
import kotlin.test.assertFailsWith

class IntegerTypeMatchesIntegersTest {
    companion object {
        val configuration = SchemaConfiguration(JsonSchemaKeywordFlags.Builder().format().build())
    }

    @Test
    fun testAnObjectIsNotAnIntegerFails() {
        // an object is not an integer
        val schema = IntegerTypeMatchesIntegers.IntegerTypeMatchesIntegers1.getInstance()
        assertFailsWith<ValidationException>(
            block = {
                schema.validate(
                    mapOf<Any?, Any?>(
                    ),
                    configuration
                )
            }
        )
    }

    @Test
    fun testAnArrayIsNotAnIntegerFails() {
        // an array is not an integer
        val schema = IntegerTypeMatchesIntegers.IntegerTypeMatchesIntegers1.getInstance()
        assertFailsWith<ValidationException>(
            block = {
                schema.validate(
                    listOf<Any?>(
                    ),
                    configuration
                )
            }
        )
    }

    @Test
    fun testNullIsNotAnIntegerFails() {
        // null is not an integer
        val schema = IntegerTypeMatchesIntegers.IntegerTypeMatchesIntegers1.getInstance()
        assertFailsWith<ValidationException>(
            block = {
                schema.validate(
                    null,
                    configuration
                )
            }
        )
    }

    @Test
    fun testAFloatWithZeroFractionalPartIsAnIntegerPasses() {
        // a float with zero fractional part is an integer
        val schema = IntegerTypeMatchesIntegers.IntegerTypeMatchesIntegers1.getInstance()
        schema.validate(
            1.0,
            configuration
        )
    }

    @Test
    fun testABooleanIsNotAnIntegerFails() {
        // a boolean is not an integer
        val schema = IntegerTypeMatchesIntegers.IntegerTypeMatchesIntegers1.getInstance()
        assertFailsWith<ValidationException>(
            block = {
                schema.validate(
                    true,
                    configuration
                )
            }
        )
    }

    @Test
    fun testAStringIsStillNotAnIntegerEvenIfItLooksLikeOneFails() {
        // a string is still not an integer, even if it looks like one
        val schema = IntegerTypeMatchesIntegers.IntegerTypeMatchesIntegers1.getInstance()
        assertFailsWith<ValidationException>(
            block = {
                schema.validate(
                    "1",
                    configuration
                )
            }
        )
    }

    @Test
    fun testAStringIsNotAnIntegerFails() {
        // a string is not an integer
        val schema = IntegerTypeMatchesIntegers.IntegerTypeMatchesIntegers1.getInstance()
        assertFailsWith<ValidationException>(
            block = {
                schema.validate(
                    "foo",
                    configuration
                )
            }
        )
    }

    @Test
    fun testAnIntegerIsAnIntegerPasses() {
        // an integer is an integer
        val schema = IntegerTypeMatchesIntegers.IntegerTypeMatchesIntegers1.getInstance()
        schema.validate(
            1,
            configuration
        )
    }

    @Test
    fun testAFloatIsNotAnIntegerFails() {
        // a float is not an integer
        val schema = IntegerTypeMatchesIntegers.IntegerTypeMatchesIntegers1.getInstance()
        assertFailsWith<ValidationException>(
            block = {
                schema.validate(
                    1.1,
                    configuration
                )
            }
        )
    }
}
